### PR TITLE
Dashboard change

### PR DESCRIPTION
**Pull request by ayushjnv1**

### CODE REVIEW

File: src/modules/dashboard/dashboard.hooks.ts [hunk 1]
```diff
@@ -3,8 +3,21 @@ import { useQuery } from 'react-query';
...
 export const useGetRequestsList = (queryParams: IFetchComplaintListRequest) => {
-  const { data, isLoading, isFetching } = useQuery('complaint-list', () =>
-    getRequestList(queryParams)
+  let path = '/tickets';
+  // if (queryParams?.department?.length > 0)
+  //   path = path + `department=${queryParams.department}`;
+  // if (queryParams?.status?.length > 0)
+  //   path = path + `category=${queryParams.}`;
+   let params = {
+    type : queryParams?.type!==""?queryParams?.type:undefined,
```
Comment (Surajrane45) [inline src/modules/dashboard/dashboard.hooks.ts:12]: Please add proper spacing for the ternary operator

File: src/modules/dashboard/dashboard.services.ts [hunk 2]
```diff
@@ -5,162 +5,163 @@ import {
...
+//     assigned_to: 'Rank',
+//     department: 'Andalax',
+//   },
+//   {
+//     id: 2,
+//     raised_by: 'Vivyanne',
+//     created_at: '3/19/2022',
+//     title: 'Angel in My Pocket',
+//     updated_time: '3/16/2022',
+//     status: 'Zathin',
```
Comment (Surajrane45) [inline src/modules/dashboard/dashboard.services.ts:26]: Remove if not required

File: src/modules/dashboard/dashboard.services.ts [hunk 3]
```diff
@@ -5,162 +5,163 @@ import {
...
+// ];
 
-export const getRequestList = (
-  queryParams: IFetchComplaintListRequest
-): Promise<GetRequestsListResponse> => {
-  return Promise.resolve({ message: 'success', data: mockData });
-  // return get({ path: '/complaints', ...queryParams });
+export const getRequestList = (path:string,params: IFetchComplaintListRequest) => {
+  //dummy
+  // return Promise.resolve({ message: 'success', data: mockData });
```
Comment (Surajrane45) [inline src/modules/dashboard/dashboard.services.ts:164]: remove commented code & { } brackets and return keyword 
we can use arrow function as 
export const getRequestList = (path:string,params: IFetchComplaintListRequest) =>get({ path,queryParams:params });

File: src/modules/dashboard/index.tsx [hunk 4]
```diff
@@ -83,55 +80,95 @@ const Dashboard = () => {
     setRowsPerPage(parseInt(event.target.value, 10));
     setPage(0);
   };
+  const { data: categoriesList, isLoading: listFetching } =
+  useCategories(departmentId);
+  const categoryOptions = useMemo(() => {
```
Comment (Surajrane45) [inline src/modules/dashboard/index.tsx:85]: keep emty line between two functions or state declaration

File: src/modules/dashboard/index.tsx [hunk 5]
```diff
@@ -83,55 +80,95 @@ const Dashboard = () => {
...
-          />
-          <CustomSelect
-            label={'Departments'}
-            options={departmentOptions}
+          />}
+           {userAuth.role !== "employee" &&  <CustomSelect
+            label={'departments'}
+            options={deptOptions}
             value={filters.department}
+            onChange={(e)=>{setDepartmentId(+(e.target.value));
```
Comment (Surajrane45) [inline src/modules/dashboard/index.tsx:123]: create separate function here & avoid inline functions

Comment (ayushjnv1) [reply src/modules/dashboard/index.tsx:123]: here we want to convert into string -> integer (only for this case )

**Comment by ayushjnv1** [comment]
changes are done

**Pull request merged**